packages/connectivity/src/scp-cf/destination/destination-from-registration.spec.ts

```diff
@@ -261,10 +262,36 @@ describe('register-destination without xsuaa binding', () => {
 
   it('registers destination and retrieves it with JWT', async () => {
     await registerDestination(testDestination, { jwt: providerServiceToken });
-    const actual = await getDestination({
-      destinationName: testDestination.name,
-      jwt: providerServiceToken
-    });
-    expect(actual).toEqual(testDestination);
+    expect(
+      await getDestination({
+        destinationName: testDestination.name,
+        jwt: providerServiceToken
+      })
+    ).toEqual(testDestination);
+  });
+
+  it('logs an error if there is a JWT, but no `zid`', async () => {
+    const logger = createLogger('register-destination');
+    jest.spyOn(logger, 'error');
+    await registerDestination(testDestination, { jwt: signedJwt({}) });
+    expect(logger.error).toHaveBeenCalledWith(
+      'Could neither determine tenant from JWT nor service binding to XSUAA, although a JWT was passed. Destination will be registered without tenant information.'
+    );
+  });
+
+  it('registers destination with a dummy id, if no JWT is given', async () => {
+    const logger = createLogger('register-destination');
+    jest.spyOn(logger, 'debug');
+
+    await registerDestination(testDestination);
+    expect(
+      await getDestination({
```

Comment:
[nitpick] Checking the registerDestinationCache.destination.retrieveDestination.. with the said dummy id instead of calling getDestination here be simpler and more explicit what is being tested.